### DISCUSSION TITLE
how to Install tensorflow package to use Embedding IDF.

### DISCUSSION THREAD

**Discussion by xianfanfen** [asreview/asreview#1653] [Q&A]
Hi!
I've encountered an issue with a missing file, specifically **`embedding-idf_feature_matrix.npz`**, even though the folder exists.

I've successfully installed the TensorFlow packages after a bit of a journey (no errors or warnings, finally!). However, during the warm-up of the AI, a hiccup occurred. It complained about not finding a file named **`embedding-idf_feature_matrix.npz`** in the directory `.asreview/e25b21dd251f41ae837d2b08fdd8e065/feature_matrices/` and told me I should install tensorflow to use Embedding IDF.

For context, here's the condensed journey:
**System**: mac os
**Path**: Set the path to `.../Python.framework/Versions/3.12/lib/python3.12/site-packages/asreview/webapp` since the default path was problematic: ``Preparing metadata (setup.py) ... error; 
  error: subprocess-exited-with-error``
After I change the path, it finally move forward. 
**Version:** Encountered version conflicts and resolved them with `pip3 install --upgrade asreview[tensorflow] --no-deps`, successfully installing TensorFlow 1.5.

I'm puzzled. Any insights into where I might be missing a step or what might be causing this hiccup?

Cheers!

**Comment by J535D165** [comment]
Hi @xianfanfen, can you try to downgrade Python to 3.12? As far as I know, Tensorflow still fails to add proper support for Python 3.12. For some users, downgrading resolves the issue. See supported Python version at https://www.tensorflow.org/install

**Reply by J535D165** [reply]
Downgrade to Python 3.11
